subprojects/packagefiles/theora/meson.build

```diff
@@ -0,0 +1,269 @@
+project(
+  'theora',
+  'c',
+  'cpp',
+  version: '1.1.1',
+  meson_version: '>= 0.59.0',
+  license: 'BSD-3-Clause',
+  default_options: ['c_std=c11'],
+)
+
+conf = configuration_data()
+
+enable_valgrind = get_option('valgrind-testing').allowed()
+
+test_env = []
+
+if enable_valgrind
+  valgrind_exe = find_program('valgrind')
+
+  conf.set('HAVE_VALGRIND', valgrind_exe.found())
+
+  if conf.get('HAVE_VALGRIND')
+    test_env += [
+      '-q',
+      '--leak-check=yes',
+      '--show-reachable=yes',
+      '--num-callers=100',
+    ]
+
+    message(
+      'Tests will be run under: @0@'.format(env.get('VALGRIND_ENVIRONMENT')),
+    )
+  else
+    message('Valgrind not found')
+  endif
+else
+  message('Valgrind testing not enabled')
+endif
+
+compile_args = []
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() != 'msvc'
+  compile_args += cc.get_supported_arguments('-Wall', '-Wno-parentheses')
+endif
+
+m_dep = cc.find_library('m', required: false)
+
+if get_option('collect-metrics').enabled()
+  add_project_arguments('-DOC_COLLECT_METRICS', language: 'c')
+endif
+
+ogg_dep = dependency(
+  'ogg',
+  version: '>= 1.1',
+  required: true,
+  not_found_message: '''
+    libogg is required to build this package!
+    please see http://www.xiph.org/ for how to
+    obtain a copy.
+  ''',
+)
+
+vorbis_dep = dependency('vorbis', version: '>= 1.0.1', required: false)
+
+sdl_dep = dependency('sdl', required: false)
+
+if not sdl_dep.found()
+  message('*** Unable to find SDL -- Not compiling example players ***')
+endif
+
+if (
+  cc.has_header('sys/soundcard.h')
+  or cc.has_header('soundcard.h')
+  or cc.has_header('machine/soundcard.h')
+)
+  if host_machine.system() == 'openbsd'
+    oss_dep = dependency('ossaudio', required: false, disabler: true)
+  else
+    oss_dep = declare_dependency()
+  endif
+else
+  oss_dep = disabler()
+endif
+
+if not oss_dep.found()
+  message('OSS audio support not found -- not compiling player_example')
+endif
+
+libpng_dep = dependency('libpng', required: false)
+
+tiff_dep = dependency('tiff', required: false, disabler: true)
+
+if (
+  tiff_dep.found()
+  and not cc.has_header_symbol(
+    'tiffio.h',
+    'TIFFReadRGBAImage',
+    dependencies: tiff_dep,
+  )
+)
+  tiff_dep = disabler()
+endif
+
+telemetry = get_option('telemetry')
+
+cairo_dep = dependency('cairo', required: false)
+
+if telemetry.allowed()
+  if cairo_dep.found()
+    conf.set('HAVE_CAIRO', true)
+  else
+    message('libcairo not found -- not compiling telemetry output support ')
+  endif
+endif
```

Comment:
Seems like this should all just be:

```meson
cairo_dep = dependency('cairo', required: get_option('telemetry'))
if cairo_dep.found()
    conf.set('HAVE_CAIRO', true)
endif
```

Reply:
There's no need to emulate upstream's handling of whether or not to error out vs. disable a feature in configure.ac -- that's a bug in their configure.ac if it ignores the user's instructions. :)

But if you were going to copy configure.ac that closely, you'd use meson boolean options for everything.